common4j/src/main/com/microsoft/identity/common/java/cache/AbstractAccountCredentialCache.java

```diff
@@ -241,7 +244,14 @@ protected List<Credential> getCredentialsFilteredByInternal(@NonNull final List<
                     atType = atType.trim();
                 }
 
-                matches = matches && authScheme.equalsIgnoreCase(atType);
+                if (TokenRequest.TokenType.POP.equalsIgnoreCase(atType)) {
+                    matches = matches && (
+                            authScheme.equalsIgnoreCase(PopAuthenticationSchemeWithClientKeyInternal.SCHEME_POP_WITH_CLIENT_KEY)
+                            || authScheme.equalsIgnoreCase(PopAuthenticationSchemeInternal.SCHEME_POP)
+                    );
+                } else {
+                    matches = matches && authScheme.equalsIgnoreCase(atType);
```

Comment:
What exactly are we matching against here? Bearer?

Reply:
Bearer or Pop

Reply:
but PoP is in the `if` block....I believe only possibility for else if Bearer?